src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js

```diff
@@ -68,29 +74,131 @@ class MapViewVisualization extends Visualization {
         }
 
         let defaultMapStyle = 'mapbox://styles/mapbox/light-v9';
+        let accentColor     = [1,234,146];
         if (document.getElementById("root").classList.contains("dark")){
             defaultMapStyle = 'mapbox://styles/mapbox/dark-v9';
+            accentColor     = [222,162,47];
         }
 
+        let preparedDataPoints = []
+        let computed           = this.prepareDataPoints(parsedData,preparedDataPoints,accentColor);
+
+        const scatterplotLayer = new deck.ScatterplotLayer({
+            data: preparedDataPoints,
+            getFillColor: d => d.color,
+            getRadius: d => d.radius
+        })
+        //
+        let latitudeMatch  = parsedData.latitude !== undefined && parsedData.latitude !== null;
+        let latitude       = latitudeMatch ? parsedData.latitude : computed.latitude;
+        let longitudeMatch = parsedData.longitude !== undefined && parsedData.longitude !== null;
+        let longitude      = longitudeMatch ? parsedData.longitude : computed.longitude;
+        // TODO : Compute zoom somehow.
+        let zoomMatch      = parsedData.zoom !== undefined && parsedData.zoom !== null;
+        let zoom           = zoomMatch ? parsedData.zoom : computed.zoom;
 
         const deckgl = new deck.DeckGL({
             container: 'map',
-            mapboxApiAccessToken: 'pk.eyJ1IjoiZW5zby1vcmciLCJhIjoiY2tmNnh5MXh2MGlyOTJ5cWdubnFxbXo4ZSJ9.3KdAcCiiXJcSM18nwk09-Q',
+            mapboxApiAccessToken: TOKEN,
             mapStyle: parsedData.mapStyle || defaultMapStyle,
             initialViewState: {
-                longitude: parsedData.longitude || 0.0,
-                latitude: parsedData.latitude || 0.0,
-                zoom: parsedData.zoom || 3,
+                longitude: longitude,
+                latitude: latitude,
+                zoom: zoom,
                 pitch: parsedData.pitch || 0
             },
             controller: parsedData.controller || true
         });
 
         deckgl.setProps({
-            layers: parsedData.layers || []
+            layers: [scatterplotLayer]
+        });
+    }
+
+    /**
+     * Prepares data points to be shown on the map.
+     * @param preparedDataPoints - List holding geoPoints.
+     * @param parsedData - All the parsed data to create GeoPoints from.
+     * @param accentColor - accent color of IDE if element doesn't specify one.
+     */
+    prepareDataPoints(parsedData, preparedDataPoints, accentColor) {
```

Comment:
This name is bad. It does not tell what it really does. Neither the docs do. I mean, I would be ok with the name here in case you really, really cannot figure something better, but then the docs have to be very explicit what are the steps of the "prepare" process

Reply:
Docs were improved 👍